File: SpringBootApp/src/main/java/com/example/springbootapp/Data/Entities/WishlistItem.java
package com.example.springbootapp.Data.Entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDate;

@Entity
@Table(name = "wishlist_items", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"wishlist_id", "product_id"})
})
@Data
@NoArgsConstructor
public class WishlistItem {
    @Id
    @UuidGenerator
    private String id;

    @Column(name ="date_added", nullable = false)
    private LocalDate dateAdded;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "wishlist_id", nullable = false)
    private Wishlist wishlist;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;
}

File: ClientUser/app/src/main/java/com/example/clientuser/model/WishlistItem.kt
package com.example.clientuser.model

import com.example.clientuser.model.dto.ProductDto
import com.example.clientuser.model.dto.WishlistItemDto
import java.time.LocalDate

class WishlistItem(
    val id: String,
    val dateAdded: LocalDate,
    val product: Product
) {
    companion object{
        suspend fun fromDto(wishlistItemDto: WishlistItemDto): WishlistItem{
            return WishlistItem(
                id = wishlistItemDto.id,
                dateAdded = wishlistItemDto.dateAdded,
                product = Product.fromDto(wishlistItemDto.product)
            )
        }
    }
}